example/ck_tile/03_gemm/gemm_basic.cpp

```diff
@@ -100,30 +108,49 @@ float gemm_(const ck_tile::GemmHostArgs& args, const ck_tile::stream_config& s)
     return ave_time;
 }
 
-float gemm(const gemm_traits& t, const ck_tile::GemmHostArgs& args, const ck_tile::stream_config& s)
+template <typename DataType>
+float gemm_type_(const gemm_traits& t,
+                 const ck_tile::GemmHostArgs& args,
+                 const ck_tile::stream_config& s)
 {
     if(t.is_a_rowmajor && t.is_b_rowmajor && t.is_c_rowmajor)
     {
-        return gemm_<Row, Row, Row>(args, s);
+        return gemm_<Row, Row, Row, DataType>(args, s);
     }
     else if(t.is_a_rowmajor && !t.is_b_rowmajor && t.is_c_rowmajor)
     {
-        return gemm_<Row, Col, Row>(args, s);
+        return gemm_<Row, Col, Row, DataType>(args, s);
     }
     else if(!t.is_a_rowmajor && t.is_b_rowmajor && t.is_c_rowmajor)
     {
-        return gemm_<Col, Row, Row>(args, s);
+        return gemm_<Col, Row, Row, DataType>(args, s);
     }
     else if(!t.is_a_rowmajor && !t.is_b_rowmajor && t.is_c_rowmajor)
     {
-        return gemm_<Col, Col, Row>(args, s);
+        return gemm_<Col, Col, Row, DataType>(args, s);
     }
     else
     {
         throw std::runtime_error("Wrong! Layouts not supported!\n");
     }
 }
 
+float gemm(const gemm_traits& t, const ck_tile::GemmHostArgs& args, const ck_tile::stream_config& s)
+{
+    if(t.data_type == "fp16")
+    {
+        return gemm_type_<GemmFp16>(t, args, s);
+    }
+    else if(t.data_type == "bf16")
+    {
+        return gemm_type_<GemmBf16>(t, args, s);
+    }
+    else
+    {
+        throw std::runtime_error("Wrong! Data type not supported!\n");
+    }
```

Comment:
We should have also support for fp8. Take a look in here: https://github.com/ROCm/composable_kernel/blob/3de7bd67ef2c5d984bde4ebbdb26338ad0a56650/include/ck_tile/ops/gemm/warp/warp_gemm.hpp#L125-L158 We have fp8 MFMA instructions added.